src/Gui/BaseApplication.cpp

```diff
@@ -332,10 +335,26 @@ void BaseApplication::engineBaseInitialization() {
     // Register the TimeSystem managing time dependant systems
     Scalar dt = ( m_targetFPS == 0 ? 1_ra / 60_ra : 1_ra / m_targetFPS );
     m_engine->setConstantTimeStep( dt );
+    // Register the SkeletonBasedAnimationSystem converting loaded assets to
+    // skeletons and skinning data
+    m_engine->registerSystem( "SkeletonBasedAnimationSystem",
+                              new Ra::Engine::Scene::SkeletonBasedAnimationSystem,
+                              defaultSystemPriority );
 }
+
 void BaseApplication::engineOpenGLInitialize() {
     // initialize here the OpenGL part of the engine used by the application
     m_engine->initializeGL();
+
```

Comment:
I find the following changes very intrusive.
Can we add a state `m_skinning_enabled` in the Engine, and protect all skinning-related stuff by this state ?
This state could be set to false to disable these things (e.g. in command line applications, or any other project not related to animation).

Reply:
Base application is a QtApplication and is not used in CLI based on Radium Core only(do you have some usecases of CLI that used BaseApplication ? ). CLI that uses Engine neither use `BaseApplication` and add explicitely the needed systems. So, no intrusion of this modification other than in graphical `BaseApplication` derived apps (Sandbox ?)

For Graphical app that do not bother about animation, this add no overhead, except a very small at load time. Only when loading animated scenes AND activating the play state of the engine there generate something.
But we can envision to modify the interface of `BaseApplication::initialize` by adding a "systems" parameter so that the user can ask for the system they want. But this is out of scope of this PR.

Reply:
Still, BaseApplication and Engine are supposed to be general, and I don’t like the idea to force loading ressources for something MyApllication will never do.
If you don’t like the idea of having a state, why not initialize the texture only when the skinning is called the first time ?

Reply:
You're partly wrong. BaseApplication is clearly not general. It defines an application with a Gui (even minimalistic) and that own a viewer (an opengl windows to render the scene). So it is very specific to Gui/openGL application.
For CLI application, even those that rely on Engine, the BaseApplication class is not used.

I like the idea of asking only for needed components. But what I said is that it is **out of scope of this PR** as it will incur changes in several interfaces not related to SkeletonBasedAnimation. 

Here, the skeletonanimationsystem is only added for applications derived from BaseApplication, i.e. Graphical applications with OpenGL windows ... and there is no overhead in your application if you do not use the animationsystem. 

So yes, definitely, we need to be able to asks for the systems to be used when creating a graphical interface. But again, it is out of scope of this PR and you can open an issue/feature request on that, so that we'll remember and can solve this in another PR.

And I don't understand why you speak about texture, it is generated only after openGL initialization so that the attached systems can be used and the openGL representation of the texture is generated only at the first use. Here, we just register the texture to the texture manager. And, if you think it is not related to BaseApplication, you're right but for now, it works like this and we'll put elsewhere in the PR that manage the state.

Reply:
Could it be simpler to just add an OpenGL initialization method in the Systems?

Reply:
This is what I plan to do in a next PR, alongside the selective instanciation of systems (this might modify a lot of things)

Reply:
I tried to fix this and it is not so direct. So, This will be solved in another PR.